agent/php_internal_instrument.c

```diff
@@ -1531,6 +1533,71 @@ NR_INNER_WRAPPER(memcache_function) {
                                              INTERNAL_FUNCTION_PARAM_PASSTHRU);
 }
 
+NR_INNER_WRAPPER(memcached_add_server) {
+  char* host = NULL;
+  nr_string_len_t host_len = 0;
+  zend_long port = 0;
+  zend_long weight = 0;
+  nr_datastore_instance_t* instance = NULL;
+  char* instance_metric = NULL;
+  int zcaught = 0;
+
+  if (SUCCESS
+      == zend_parse_parameters_ex(
+          ZEND_PARSE_PARAMS_QUIET, ZEND_NUM_ARGS(), "s|ll", &host,
+          &host_len, &port, &weight) &&
+      NULL != host) {
+    instance = nr_php_memcached_create_datastore_instance(host, port);
```

Comment:
does creating it as an `instance` format the host and/or path in a special way?
Can we just pass host and port directly to the [nr_format](https://github.com/newrelic/newrelic-php-agent/pull/958/files#diff-d66d558f126dbc42fe98d5332fa13d281fe925d4e01b7608a8dc39c56dce778eR1551) command?  ten we wouldn't have to create/free the `instance`.

Reply:
Oh, I see, [it is a little bit of special checking](https://github.com/newrelic/newrelic-php-agent/pull/958/files#diff-50b3bcc41a255bfafa2a5b71bdf4d9ec0d8165f881ddeb9bbeb0118f2d31c843R10).